### PR TITLE
Traffic graphs

### PR DESCRIPTION
**Pull request by somospocos**
To see how it works, run the traffic module with something like **-p traffic.graphlen=10**

### CODE REVIEW

File: bumblebee/modules/traffic.py [hunk 1]
```diff
@@ -120,10 +140,19 @@ def _update_widgets(self, widgets):
...
-                    (int(data[direction]) - int(prev))/timediff,
-                    self._format)
+                bspeed = (int(data[direction]) - int(prev))/timediff
+                speed = bumblebee.util.bytefmt(bspeed, self._format)
                 txtspeed = '{0}/s'.format(speed)
+                if self._graphlen > 0:
+                    # skip first value returned by psutil, because it is
+                    # giant and ruins the grapth ratio until it gets pushed
+                    # out of saved list
+                    if self._first_run is True:
```
Comment (tobi-wan-kenobi) [inline bumblebee/modules/traffic.py]: maybe it would be more elegant to check for len(self._graphdata[interface][direction]) > 1 ?

**Comment by tobi-wan-kenobi** [comment]
Very cool addition, thanks!

**Comment by tobi-wan-kenobi** [comment]
@somospocos I want to thank you for the many additions you've made for bumblebee-status recently, those are really big improvements that I appreciate a lot.

I just wanted to let you know two things, and I don't see any other way to contact you, except for this.

1. For personal reasons, I only have a very limited amount of time per week for this project, at the moment. So please bear with me if reviewing a PR, or responding to a comment, takes a while.

2. Having said that, I want to clean up and simplify (=rewrite) the bumblebee core components, and achieve a higher test coverage, at the same time. This is likely a multi-month effort that I have started in the `development` branch, but I thought I'd let you know, in case you notice the branch. It will definitely also incorporate all your improvements to the core (braille, charts), but I might have to modify the implementation a bit to make them fit. (And, TBH, let's see whether I have sufficient time to even finish that effort).

So, once again: Thanks! :)

**Comment by somospocos** [comment]
> maybe it would be more elegant to check for len(self._graphdata[interface][direction]) > 1 ?

You are welcome to refactor anything for improving aesthetics. The length of those data lists is always constant and equal to traffic.graphlen, it does not vary. I only move data inside those lists, I don't change their size.

> I want to thank you for the many additions you've made for bumblebee-status recently, those are really big improvements that I appreciate a lot.

Not a big deal, I was just scratching the personal itches that I raised here https://github.com/tobi-wan-kenobi/bumblebee-status/issues/486#issue-538468429 :) I was lucky I knew a bit of Python and that the codebase has a good design and was easy to work with - this is entirely your merit, thank you. If these additions end being useful to others as well - everybody wins.

> For personal reasons, I only have a very limited amount of time per week for this project, at the moment. So please bear with me if reviewing a PR, or responding to a comment, takes a while.

By coincidence, this was the last "big" thing that I was missing. It was great that you could accept a couple of pull requests today very quick after they were opened. Now that I feel like I have all the building blocks I wanted, I can finally slow down and proceed to building my custom config. This means that in the near future there will be no more stress for you about having a pull request that's waiting for a couple of days :)

> It will definitely also incorporate all your improvements to the core (braille, charts), but I might have to modify the implementation a bit to make them fit.

As I said above, feel free to refactor anything you want to.

**Pull request merged**